File: src/main/java/org/cip4/xjdf/json/openapi/model/OpenApi.java
package org.cip4.xjdf.json.openapi.model;

import com.fasterxml.jackson.annotation.JsonTypeName;
import lombok.*;

import java.util.Map;

@AllArgsConstructor
@Getter
@JsonTypeName("OpenApi")
public class OpenApi {
    @NonNull
    private final String openapi;
    @NonNull
    private final Info info;
    @NonNull
    private final Map<String, PathItem> paths;
    @NonNull
    private final Components components;
}

File: src/main/kotlin/org/cip4/xjdf/openapi/model/OpenApi.kt
package org.cip4.xjdf.openapi.model

import com.charleskorn.kaml.Yaml
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString

@Serializable
@SerialName("OpenApi")
data class OpenApi(
    val openapi: String,
    val info: Info,
    val paths: Map<String, PathItem>,
    var components: Components = Components()
) : YmlModel {
    override fun toYml(): String {
        return Yaml.default.encodeToString(this)
    }
}